lease/leasehttp/http.go

```diff
@@ -59,6 +71,13 @@ func (h *leaseHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "error unmarshalling request", http.StatusBadRequest)
 			return
 		}
+		cci := h.i.GetCommittedIndex()
+		select {
+		case <-h.w.Wait(cci):
+		case <-time.After(applyTimeout):
+			http.Error(w, "waiting for node to catch up its applied index has timed out", http.StatusInternalServerError)
```

Comment:
should be an actual error code like the other errors; otherwise, this will cause the client lease to shutdown despite this most likely being a transient error